File: Gregs2/Models/House.cs
using Gregs2.interfaces;
namespace Gregs2.Models;

public class House : IHasCreator
{
  public int Id { get; set; }
  public string CreatorId { get; set; }
  public Profile Creator { get; set; }
  public DateTime CreatedAt { get; set; }
  public DateTime UpdatedAt { get; set; }
  public int Bedrooms { get; set; }
  public int Bathrooms { get; set; }
  public int Levels { get; set; }
  public string ImgUrl { get; set; }
  public int Price { get; set; }
  public string Description { get; set; }
  public bool IsSold { get; set; }
}
public class FavoritedHouse : House
{
  public int FavoriteId { get; set; }
}

File: Gregs2.client/src/models/House.js
export class House {
  constructor(data) {
    this.Id = data.Id
    this.CreatorId = data.CreatorId
    this.Creator = data.Creator
    this.CreatedAt = data.CreatedAt
    this.UpdatedAt = data.UpdatedAt
    this.Bedrooms = data.Bedrooms
    this.Bathrooms = data.Bathrooms
    this.Levels = data.Levels
    this.ImgUrl = data.ImgUrl
    this.Price = data.Price
    this.Description = data.Description
    this.IsSold = data.IsSold
    this.FavoriteId = data.FavoriteId
  }
}